airbyte-webapp/src/components/connectorBuilder/Builder/SavingIndicator.tsx

```diff
@@ -0,0 +1,65 @@
+import { faCheck } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { Button } from "components/ui/Button";
+import { FlexContainer, FlexItem } from "components/ui/Flex";
+import { Spinner } from "components/ui/Spinner";
+import { Text } from "components/ui/Text";
+import { Tooltip } from "components/ui/Tooltip";
+
+import { SavingState, useConnectorBuilderFormState } from "services/connectorBuilder/ConnectorBuilderStateService";
+
+import styles from "./SavingIndicator.module.scss";
+
+function getMessage(savingState: SavingState, triggerUpdate: () => void) {
+  if (savingState === "invalid") {
+    return (
+      <Tooltip
+        placement="bottom-start"
+        control={
+          <span className={styles.invalid}>
+            <FormattedMessage id="connectorBuilder.loadingState.invalid" />
+          </span>
+        }
+      >
+        <FormattedMessage id="connectorBuilder.loadingState.invalid.tooltip" />
+      </Tooltip>
+    );
+  }
+  if (savingState === "loading") {
+    return (
+      <FlexContainer gap="sm" alignItems="center">
+        <Spinner size="xs" />
+        <FormattedMessage id="connectorBuilder.loadingState.loading" />
+      </FlexContainer>
+    );
+  }
+  if (savingState === "error") {
+    return (
+      <Button variant="clear" onClick={triggerUpdate}>
```

Comment:
UX nit: I tested this state out by shutting down my local docker compose deployment, then making a change in the builder UI. In this state, when I clicked this button to retry, there was no UX feedback for me that anything happened, likely because the request failed so quickly that the `loading` state didn't have time to be visible. 

This probably isn't worth spending much time on, but maybe we could force this to stay in `loading` for some amount of time, say 200ms, so that the user at least sees that something was tried. What do you think?